crates/bevy_asset/src/io/android.rs

```diff
@@ -46,15 +45,53 @@ impl AssetReader for AndroidAssetReader {
 
     async fn read_directory<'a>(
         &'a self,
-        _path: &'a Path,
+        path: &'a Path,
     ) -> Result<Box<PathStream>, AssetReaderError> {
-        let stream: Box<PathStream> = Box::new(EmptyPathStream);
-        error!("Reading directories is not supported with the AndroidAssetReader");
-        Ok(stream)
+        let asset_manager = bevy_winit::ANDROID_APP
+            .get()
+            .expect("Bevy must be setup with the #[bevy_main] macro on Android")
```

Comment:
Can we add this expectation to the documentation?

Reply:
This is a message that is also used in `bevy_winit`, so it is just reused here. I agree it would be good to add to the docs. I can add it separately to the `android` page in the book (which is currently hidden, but I guess it is good to have it there so it is not forgotten once it gets published).